Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectILLEnergyTransfer.py

```diff
@@ -592,26 +621,48 @@ def _reduce_bats(self, ws):
             DeleteWorkspace(equator_ws)
 
             if self._fit_option == 'FitAllPixelGroups':
-                GroupDetectors(InputWorkspace=ws, OutputWorkspace=grouped_ws, GroupingPattern=self._group_pixels(self._group_by))
+                GroupDetectors(InputWorkspace=ws,
+                               OutputWorkspace=grouped_ws,
+                               GroupingPattern=self._group_pixels(self._group_by, single_detectors))
                 CropWorkspace(InputWorkspace=grouped_ws, OutputWorkspace=grouped_ws, XMin=to_crop, XMax=3 * to_crop)
                 FindEPP(InputWorkspace=grouped_ws, OutputWorkspace=epp_ws)
-                self._create_elastic_channel_ws(mtd[epp_ws], mtd[ws].getRun(), mtd[equator_epp_ws])
+                self._create_elastic_channel_ws(mtd[epp_ws], mtd[ws].getRun(), mtd[equator_epp_ws],
+                                                single_detectors=single_detectors)
                 DeleteWorkspaces([equator_epp_ws, grouped_ws])
             else:
-                self._create_elastic_channel_ws(mtd[equator_epp_ws], mtd[ws].getRun())
-                RenameWorkspace(InputWorkspace=equator_epp_ws, OutputWorkspace=epp_ws)
+                single_det_ws = _make_name(ws, "sds")
+                ExtractSpectra(InputWorkspace=ws,
+                               OutputWorkspace=single_det_ws,
+                               StartWorkspaceIndex=N_TUBES*N_PIXELS_PER_TUBE + 1,
+                               EndWorkspaceIndex=N_TUBES*N_PIXELS_PER_TUBE + single_detectors)
+                FindEPP(InputWorkspace=single_det_ws, OutputWorkspace=epp_ws)
+                self._create_elastic_channel_ws(mtd[epp_ws], mtd[ws].getRun(), mtd[equator_epp_ws],
```

Comment:
This is a bit confusing, are you passing the epp of single detectors as the first argument, together with number of single detectors? If so, maybe clarify in the doxygen comments of _create_elastic...